Constructive Code Review/outline.txt

```diff
@@ -0,0 +1,205 @@
+          or angry.
+            - Why am I feeling that way? Sometimes that reveals things
+              I believe that aren't true or things I need to think
+              about.
+            - I generally make bad decisions and say things I regret
+              when feeling that way.
+            - It's a good time to talk a break and come back with a
+              clear head.
+    - Feeling short on time
+          There will always be more to review than can be reviewed. So
+          we have to learn to adapt to that reality: lower standards,
+          never sleep, or just go at a sustainable pace. It can do a
+          lot for a contrib's peace of mind just to know you've seen
+          that the patch is there and that you haven't forgotten about
+          it. Sometimes I even let people know "It's next in my queue."
+        - Consider: what's the risk if there's something wrong with it?
+            - Will it cause a lot of damage?
+            - Will it be easy to fix, or is there, say, a data-exchange
+              format being defined?
+    - Interpersonal issues
+        - When you're in person with your team, make a point of
+          establishing good relations with them.
+            - Then when you have to critique each other or talk on a
+              low-bandwidth medium like IRC, you'll know how to take
+              each other. You'll have some built-up offense-buffering.
+    - Other feelings
+        - When you can't decide what to say (something feels too bad to
+          merge), you can always fall back to articulating your
+          emotions: "I feel uncomfortable about leaving plain-text
+          passwords on disk."
+        - You don't have to be definitive, decisive, or an authority
+          all the time. Just treat everybody like adults and articulate
+          your feelings. I've said "I'm happy you want to work on that,
+          but I'm afraid that you'll end up in merge hell if I land the
+          elasticsearch stuff first." Inviting them into the tradeoff
+          is a heckuva lot better than "no!". (If they don't mine merge
+          hell, they can go for it!)
+- Dealing with newbies
+      We’ve all got to start where we’re at. Don’t make fun of scrawny
+      people weightlifting.
+    - I don't spend much time new contribs the moment they show up.
+        - But my investment parallels theirs.
+    - Leveling up
+        - They don't know where to start
+            - Links to dev docs, a few sentences of explanation and
+              ideas where to start
+            - Links to resources to learn skills if they don't have them
+        - First I will fix their patch (geared for one-timers so
+          patches don't rot)
+            - "Polish-up" commits
+                - So their name still gets into the commit log and they
+                  feel good
+        - Next, I just comment and request fixes
+        - Then I require tests
+- Call to action
```

Comment:
So... you should treat yourself to a piece of cake!

Argh. I have no idea what happened to the rest of that comment. I think it got eaten by the ravenous bugblatter beast.